From 01dcc3eec868a3e6529c9c4733da628428763a07 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Fri, 14 Jun 2013 13:38:02 +0200
Subject: [PATCH 0607/1083] drm: Improve manual IRQ installation documentation

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit f57c9646612678af80766013cd88f43df6866187)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 3950306fbbf8d6b18b1b144b7e91af9aa44182a1)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 Documentation/DocBook/drm.tmpl | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/Documentation/DocBook/drm.tmpl b/Documentation/DocBook/drm.tmpl
index 4ee2304..7e1191c 100644
--- a/Documentation/DocBook/drm.tmpl
+++ b/Documentation/DocBook/drm.tmpl
@@ -186,11 +186,12 @@
           <varlistentry>
             <term>DRIVER_HAVE_IRQ</term><term>DRIVER_IRQ_SHARED</term>
             <listitem><para>
-              DRIVER_HAVE_IRQ indicates whether the driver has an IRQ handler. The
-              DRM core will automatically register an interrupt handler when the
-              flag is set. DRIVER_IRQ_SHARED indicates whether the device &amp;
-              handler support shared IRQs (note that this is required of PCI
-              drivers).
+              DRIVER_HAVE_IRQ indicates whether the driver has an IRQ handler
+	      managed by the DRM Core. The core will support simple IRQ handler
+	      installation when the flag is set. The installation process is
+	      described in <xref linkend="drm-irq-registration"/>.</para>
+	      <para>DRIVER_IRQ_SHARED indicates whether the device &amp; handler
+	      support shared IRQs (note that this is required of PCI  drivers).
             </para></listitem>
           </varlistentry>
           <varlistentry>
@@ -344,7 +345,8 @@ char *date;</synopsis>
           The DRM core tries to facilitate IRQ handler registration and
           unregistration by providing <function>drm_irq_install</function> and
           <function>drm_irq_uninstall</function> functions. Those functions only
-          support a single interrupt per device.
+          support a single interrupt per device, devices that use more than one
+	  IRQs need to be handled manually.
         </para>
   <!--!Fdrivers/char/drm/drm_irq.c drm_irq_install-->
         <para>
-- 
1.8.3.2

